components/event-publisher/event-output-adapters/org.wso2.carbon.event.output.adapter.email/src/main/java/org/wso2/carbon/event/output/adapter/email/EmailEventAdapter.java

```diff
@@ -95,22 +116,123 @@ public void testConnect() throws TestConnectionNotSupportedException {
         throw new TestConnectionNotSupportedException("not-available");
     }
 
+    /**
+     * Initialize the Email SMTP session and be ready to send emails.
+     *
+     * @throws ConnectionUnavailableException on error.
+     */
+
+
     @Override
-    public void connect() {
-        //not required
+    public void connect() throws ConnectionUnavailableException {
+
+        if (session == null) {
+
+            /**
+             * Default SMTP properties for outgoing messages.
+             */
+            String smtpFrom;
+            String smtpHost;
+            String smtpPort;
+
+
+            /**
+             *  Default from username and password for outgoing messages.
+             */
+
+            final String smtpUsername;
+            final String smtpPassword;
+
+
+            // initialize SMTP session.
+            Properties props = new Properties();
+            props.putAll(globalProperties);
+
+            //Verifying default SMTP properties of the SMTP server.
+
+            smtpFrom = props.getProperty(MailConstants.MAIL_SMTP_FROM);
+            smtpHost = props.getProperty(EmailEventAdapterConstants.MAIL_SMTP_HOST);
+            smtpPort = props.getProperty(EmailEventAdapterConstants.MAIL_SMTP_PORT);
+
+            if (smtpFrom == null) {
+                String msg = "failed to connect to the mail server due to null smtpFrom value";
+                throw new ConnectionUnavailableException("The adapter " +
+                        eventAdapterConfiguration.getName() + " " + msg);
+
+            }
+
+            if (smtpHost == null) {
+                String msg = "failed to connect to the mail server due to null smtpHost value";
+                throw new ConnectionUnavailableException
+                        ("The adapter " + eventAdapterConfiguration.getName() + " " + msg);
+            }
+
+            if (smtpPort == null) {
+                String msg = "failed to connect to the mail server due to null smtpPort value";
+                throw new ConnectionUnavailableException
+                        ("The adapter " + eventAdapterConfiguration.getName() + " " + msg);
+            }
+
+
+            try {
+                smtpFromAddress = new InternetAddress(smtpFrom);
+            } catch (AddressException e) {
+                log.error("Error in retrieving smtp address : " +
+                        smtpFrom, e);
+                String msg = "failed to connect to the mail server due to error in retrieving " +
+                        "smtp from address";
```

Comment:
Is there any specific reason for doing both logging the exception and throwing ?